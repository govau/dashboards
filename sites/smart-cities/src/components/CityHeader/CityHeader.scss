@import '~style/main';

.wrapper {
  position: relative;
  height: 120px;

  @include for-tablet-portrait-up {
    height: 160px;
  }

  @include for-tablet-landscape-up {
    height: 240px;
  }

  @include for-desktop-up {
    height: 288px;
  }
}

.container {
  @include container;

  position: relative; // above the absolute background panels
  display: flex;
  align-items: flex-end;
  padding: 16px;
  height: 100%;

  @include for-tablet-portrait-up {
    padding: 20px;
  }

  @include for-tablet-landscape-up {
    padding: 32px;
  }
}

.photoWrapper {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  overflow: hidden;
}

.photo {
  height: 100%;
  width: auto;
  display: block;
  margin: 0 auto;

  // For browsers that support object-fit (and @supports)
  // set the width to 100%
  @supports (object-fit: cover) {
    object-fit: cover;
    width: 100%;
  }
}

.title {
  line-height: 1; // it wraps on small screens
  font-size: 36px;
  font-weight: 300;
  letter-spacing: 0.7px;

  @include for-tablet-portrait-up {
    font-size: 44px;
    font-weight: 300;
  }

  @include for-desktop-up {
    font-size: 72px;
    font-weight: 300;
  }
}
