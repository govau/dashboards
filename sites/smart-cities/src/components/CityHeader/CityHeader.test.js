import React from 'react';
import { shallow } from 'enzyme';
import CityHeader from './CityHeader';

jest.mock('../Icon/Icon', () => '../Icon/Icon');

const defaultProps = {
  cityName: 'Greater perth',
  categoryColorName: 'jobs',
  cityId: 'greater-perth',
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityHeader {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should add an image with the city id', () => {
  const component = shallow(
    <CityHeader {...defaultProps} />
  );

  expect(component.find('.photo').prop('src')).toBe('/images/greater-perth.jpg');
});
