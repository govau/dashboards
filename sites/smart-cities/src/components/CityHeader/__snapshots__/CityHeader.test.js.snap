// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\" style={{...}}>
  <div className=\\"photoWrapper\\">
    <img className=\\"photo\\" src=\\"/images/greater-perth.jpg\\" alt=\\"\\" />
  </div>
  <div className=\\"container\\">
    <h1 className=\\"title\\">
      Greater perth
    </h1>
  </div>
</div>"
`;
